src/Microsoft.AspNetCore.Mvc.ApiExplorer/ApiBehaviorApiDescriptionProvider.cs

```diff
@@ -42,6 +43,40 @@ public void OnProvidersExecuting(ApiDescriptionProviderContext context)
                 {
                     description.SupportedResponseTypes.Add(responseType);
                 }
+
+                AddRequestTypeForFormFiles(description);
+            }
+        }
+
+        private void AddRequestTypeForFormFiles(ApiDescription description)
+        {
+            if (description.SupportedRequestFormats.Count > 0)
+            {
+                // If there are already supported formats registered, don't do anything.
+                return;
+            }
+
+            if (description.ParameterDescriptions.Any(p => p.Source == BindingSource.FormFile))
```

Comment:
`DefaultApiDescriptorProvider` only looks at `FromBody` parameters and then looks up their formatters. Had to replicate stuff here for files.

Reply:
Should we do this in the default provider?

Reply:
This just makes it possible for consumes to set a content type for forms, I don't feel like that's a new or surprising behavior, just a general enhancement

Reply:
Sure.